src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -294,6 +299,10 @@ protected MqttClient(DeviceConfiguration deviceConfiguration, Spool spool, Sched
     @SuppressWarnings("PMD.CloseResource")
     public synchronized void subscribe(SubscribeRequest request)
             throws ExecutionException, InterruptedException, TimeoutException {
+        if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
+            throw new ExecutionException(new Exception("Cannot subscribe because device is configured to run offline"));
```

Comment:
I'm not so sure that throwing an exception is the right thing to do at this point. Open to discussion, but I think it would be better to just log and return here instead of throwing.

Reply:
Changed to not throw. Consistent with how subscription failure is handled.